projects/distributed-tracing/src/shared/dashboard/widgets/trace-detail/data/trace-detail-data-source.model.ts

```diff
@@ -0,0 +1,72 @@
+import { Dictionary } from '@hypertrace/common';
+import { ARRAY_PROPERTY, Model, ModelProperty, PLAIN_OBJECT_PROPERTY } from '@hypertrace/hyperdash';
+import { EMPTY, Observable, of } from 'rxjs';
+import { mergeMap } from 'rxjs/operators';
+import { Trace, traceIdKey, traceTypeKey } from '../../../../graphql/model/schema/trace';
+import { SpecificationBuilder } from '../../../../graphql/request/builders/specification/specification-builder';
+import {
+  TraceGraphQlQueryHandlerService,
+  TRACE_GQL_REQUEST
+} from '../../../../graphql/request/handlers/traces/trace-graphql-query-handler.service';
+import { GraphQlDataSourceModel } from '../../../data/graphql/graphql-data-source.model';
+
+@Model({
+  type: 'trace-detail-data-source'
+})
+export class TraceDetailDataSourceModel extends GraphQlDataSourceModel<TraceDetailData> {
+  @ModelProperty({
+    key: 'trace',
+    required: true,
+    type: PLAIN_OBJECT_PROPERTY.type
+  })
+  public trace!: Trace;
+
+  @ModelProperty({
+    key: 'attributes',
+    type: ARRAY_PROPERTY.type
+  })
+  public attributes: string[] = [];
+
+  private readonly attributeSpecBuilder: SpecificationBuilder = new SpecificationBuilder();
+
+  public getData(): Observable<TraceDetailData> {
+    return this.query<TraceGraphQlQueryHandlerService>({
+      requestType: TRACE_GQL_REQUEST,
+      traceType: this.trace[traceTypeKey],
+      traceId: this.trace[traceIdKey],
+      spanLimit: 0,
+      timeRange: this.getTimeRangeOrThrow(),
+      traceProperties: this.getTraceAttributes().map(attribute =>
+        this.attributeSpecBuilder.attributeSpecificationForKey(attribute)
+      )
+    }).pipe(mergeMap(trace => this.mapResponseObject(trace)));
+  }
+
+  protected getTraceAttributes(): string[] {
+    return ['tags', 'statusCode', 'traceId', ...this.attributes];
```

Comment:
I don't believe `traceId` exists on a regular trace - that would be returned by the default id attribute already fetched (see other comment for more context).